swent-epfl-groupe07/eduverse-android
# Feature/Implemented nextScreen, Share to functionnality, picTakenScreen, and save functionnality with MVVM

Feature: Implemented NextScreen UI, "Share to" Functionality, PicTakenScreen, and Save Functionality with MVVM
Overview: This pull request introduces several key enhancements to the Eduverse application, focusing on improving the user experience in the NextScreen UI, enabling image sharing, and ensuring alignment with the MVVM architecture. Additionally, it includes comprehensive testing to achieve high code coverage and ensure robustness.

Summary of Changes:
Owner UID Configuration:

Updated NextScreen to use dynamic owner UID retrieval from FirebaseAuth.
This enhancement aligns with PR recommendations and improves user-specific data handling.
NextScreen UI Design & Implementation:

Designed and developed the NextScreen UI with a user-friendly layout.
Elements are organized and styled for easy navigation and interaction.
Implemented a responsive layout that adapts to various screen sizes, providing a consistent experience across devices.
"Share to" Functionality:

Enabled image sharing directly from NextScreen, allowing users to share their captured photos via SMS, email, or other installed apps.
This feature leverages the FileProvider for secure file sharing.
Improved user experience by integrating Android’s native sharing options, making it easy for users to share content without leaving the app.
Comprehensive Tests for NextScreen and PicTakenScreen:

Added thorough unit and UI tests to cover NextScreen and PicTakenScreen, ensuring over 90% line coverage.
The tests verify that critical UI elements are displayed correctly and validate the new "Share to" functionality.
Utilized mock objects to simulate user interactions and API responses, enhancing test reliability and accuracy.
PicTakenScreen Save Functionality with MVVM:

Implemented MVVM architecture in PicTakenScreen for better separation of concerns, making it easier to manage and test the save functionality.
Save functionality now stores the user’s captured image along with the owner UID, using the ViewModel for business logic and data persistence.
Users can now save their photos to Firebase with improved feedback and reliability.

Testing & Validation:
UI and Unit Tests:
Extensive UI testing ensures that all interactive elements in NextScreen and PicTakenScreen behave as expected.
Validated "Share to" functionality across different sharing platforms, confirming compatibility and secure file handling.
Manual Testing:
Conducted manual testing on various Android devices to confirm layout responsiveness and functionality.
Verified that the save feature securely stores photos associated with the authenticated user and that shared images are correctly attached to external intents.

Note: These updates significantly enhance user experience and align with modern Android development practices, particularly with the MVVM architecture. The new features and tests ensure a smoother, more reliable app experience for users, preparing the Eduverse application for future scalability and feature expansions.
